e107_core/shortcodes/batch/login_shortcodes.php

```diff
@@ -151,15 +151,19 @@ function sc_login_table_rememberme($parm=null)
 
 	}
 	
+	/* example {LOGIN_TABLE_SUBMIT} */
+	/* example {LOGIN_TABLE_SUBMIT=large}  old way */  
+	/* example {LOGIN_TABLE_SUBMIT: class=btn btn-primary btn-large btn-lg pull-right} */  
 	function sc_login_table_submit($parm="") //FIXME use $frm
 	{
-		if(empty($this->userReg))
+    $class = ($parm == 'large') ? "btn-large btn-lg" : "";
+    if(empty($this->pref['user_reg']))
 		{
 			return null;
 		}
-
-		$class = ($parm == 'large') ? "btn-large btn-lg" : "";
-		return "<input class='btn btn-primary ".$class." button' type='submit' name='userlogin' value=\"".LAN_LOGIN_9."\" />";
+ 
```

Comment:
Mixing `$parm` as a string and `$parm` as an array will get messy. (especially with PHP 7.1+)  It may be better to create a new shortcode which uses `$parm` only as an array. . ie. **sc_login_table_button($parm=null)**